tests/test_utils.py

```diff
@@ -39,206 +39,6 @@
     (2, 2, 2, 2, 2, 2),
 ]
 
-# global variables and functions
-I = np.identity(2)
-X = np.array([[0, 1], [1, 0]])
-Y = np.array([[0, -1j], [1j, 0]])
-Z = np.array([[1, 0], [0, -1]])
-
-
-SWAP = np.array([[1, 0, 0, 0], [0, 0, 1, 0], [0, 1, 0, 0], [0, 0, 0, 1]])
-CNOT = np.array([[1, 0, 0, 0], [0, 1, 0, 0], [0, 0, 0, 1], [0, 0, 1, 0]])
-
-
-U = np.array(
-    [
-        [0.83645892 - 0.40533293j, -0.20215326 + 0.30850569j],
-        [-0.23889780 - 0.28101519j, -0.88031770 - 0.29832709j],
-    ]
-)
-
-U2 = np.array([[0, 1, 1, 1], [1, 0, 1, -1], [1, -1, 0, 1], [1, 1, -1, 0]]) / np.sqrt(3)
-
-
-U_toffoli = np.diag([1 for i in range(8)])
-U_toffoli[6:8, 6:8] = np.array([[0, 1], [1, 0]])
-
-
-class TestSparse:
-    """Tests the sparse_hamiltonian function"""
```

Comment:
I'm sure `sparse_matrix` is tested in general, but do we have equivalents for these particular tests already?

Reply:
Yep. unit tests for `Hamiltonian.sparse_matrix`.